### PR TITLE
Fast load conversationsummarymemory from existing summary

### PR DESCRIPTION
**Pull request by ShantanuNair**
- Description: Adds an optional buffer arg to the memory's from_messages() method. If provided the existing memory will be loaded instead of regenerating a summary from the loaded messages.
 
Why? If we have past messages to load from, it is likely we also have an existing summary. This is particularly helpful in cases where the chat is ephemeral and/or is backed by serverless where the chat history is not stored but where the updated chat history is passed back and forth between a backend/frontend. 

Eg: Take a stateless qa backend implementation that loads messages on every request and generates a response — without this addition, each time the messages are loaded via from_messages, the summaries are recomputed even though they may have just been computed during the previous response. With this, the previously computed summary can be passed in and avoid:
  1) spending extra $$$ on tokens, and 
  2) increased response time by avoiding regenerating previously generated summary.

Tag maintainer: @hwchase17
Twitter handle: https://twitter.com/ShantanuNair

### CODE REVIEW

File: langchain/memory/summary.py [hunk 1]
```diff
@@ -48,13 +48,18 @@ def from_messages(
         chat_memory: BaseChatMessageHistory,
         *,
         summarize_step: int = 2,
+        existing_summary: str = "",
```
Comment (baskaryan) [inline langchain/memory/summary.py:51]: currently someone could already pass in buffer as part of **kwargs and that'd be passed in at instantiation, but then overwritten by either existing_summary or messages, which is kinda confusing.

what if we did
```python
  @classmethod
  def from_messages(
      cls,
      llm: BaseLanguageModel,
       chat_memory: BaseChatMessageHistory,
       *,
       summarize_step: int = 2,
       buffer: str = "",
       **kwargs:  Any
   ) -> ConversationSummaryMemory:
     obj = cls(llm=llm, chat_memory=chat_memory, buffer=buffer, **kwargs)
     if not buffer:
       for i in range(0, len()...)):
         obj.buffer = obj.predict_new_summary(...)
     return obj

Comment (ShantanuNair) [reply langchain/memory/summary.py]: @baskaryan Makes sense! I updated the PR to use the buffer arg.

**Comment by ShantanuNair** [comment]
@baskaryan @hwchase17

File: langchain/memory/summary.py (hunk 1 above)
Comment (ShantanuNair) [reply langchain/memory/summary.py]: @baskaryan Resolved this

**Review by hwchase17 — APPROVED** [review]
updated to just show how to initialize directly

**Pull request merged**